soc/intel/intel_adsp/ace/power.c

```diff
@@ -456,6 +471,13 @@ void arch_cpu_idle(void)
 	}
 
 	__asm__ volatile ("waiti 0");
+#if CONFIG_ADSP_DYNAMIC_CLOCK_SWITCHING
+	k = k_spin_lock(&idle_lock);
+	if (data && intel_adsp_acquire_clock_switch_lock(data))
```

Comment:
What are the circumstances where data might be NULL?  This seems like a needless check to me that could be better expressed as an assert?

Also the MISRA scanner is snooping around again and it doesn't like the "implicitly convert pointer to boolean" idiom and demands this be expressed as "data == NULL".  But maybe it won't find its way into the soc code for a while.